ddl/ddl_api.go

```diff
@@ -7475,6 +7475,11 @@ func BuildAddedPartitionInfo(ctx sessionctx.Context, meta *model.TableInfo, spec
 		if len(spec.PartDefinitions) == 0 {
 			return nil, ast.ErrPartitionsMustBeDefined.GenWithStackByArgs(meta.Partition.Type)
 		}
+		err := checkListPartitions(ctx, spec.PartDefinitions)
```

Comment:
From https://dev.mysql.com/worklog/task/?id=5781:

> When using a DEFAULT partition, it will not longer be possible to ADD a partition, but one must instead REORGANIZE the DEFAULT partition, just like MAXVALUE partition in RANGE partitioning.

Now we allow `ADD PARTITION` without reorganize the data, is it expected?

Reply:
I added a check for that `checkAddListPartitions`.

Reply:
Great catch btw!